src/nvim/undo_defs.h

```diff
@@ -2,13 +2,42 @@
 
 #include <time.h>
 
+#include "nvim/block_defs.h"
+#include "nvim/eval/typval_defs.h"
 #include "nvim/extmark_defs.h"
 #include "nvim/mark_defs.h"
+#include "nvim/normal_defs.h"
 
 enum { UNDO_HASH_SIZE = 32, };  ///< Size in bytes of the hash used in the undo file.
 
 typedef struct u_header u_header_T;
 
+typedef struct {
+  u_header_T *save_b_u_oldhead;
+  u_header_T *save_b_u_newhead;
+  u_header_T *save_b_u_curhead;
+  int save_b_u_numhead;
+  bool save_b_u_synced;
+  int save_b_u_seq_last;
+  int save_b_u_save_nr_last;
+  int save_b_u_seq_cur;
+  time_t save_b_u_time_cur;
+  int save_b_u_save_nr_cur;
+  char *save_b_u_line_ptr;
+  linenr_T save_b_u_line_lnum;
+  colnr_T save_b_u_line_colnr;
+} UndoInfo;
+
+typedef struct {
```

Comment:
a one-line docstring that describes this would be useful. I assume it's `preview buffer info`

Reply:
I'll add it